packages/date/src/index.js

```diff
@@ -426,9 +549,22 @@ export function dateI18n( dateFormat, dateValue = new Date(), timezone ) {
 		timezone = undefined;
 	}
 
-	const dateMoment = buildMoment( dateValue, timezone );
-	dateMoment.locale( settings.l10n.locale );
-	return format( dateFormat, dateMoment );
+	return formatTZ(
+		utcToZonedTime(
```

Comment:
Hi @vcanales, We are passing a manipulated date value to formatTZ, and the original value to translateFormat. translateFormat for some formats like 'c' outputs the date directly. I think the date value we pass to formatTZ and to translateFormat should always be the same, otherwise depending on if it is a mapped format or a format that is a function we output different things. In PR https://github.com/WordPress/gutenberg/pull/27272/ I added unit tests that show this bug happening.